Python/specialize.c

```diff
@@ -2663,101 +2663,98 @@ _Py_Specialize_Send(_PyStackRef receiver_st, _Py_CODEUNIT *instr)
     cache->counter = adaptive_counter_cooldown();
 }
 
+static int
```

Comment:
This should be surrounded by `#ifdef Py_STATS`. We don't want to be calling this unless stats are enabled.

Reply:
@mpage 
I have same concerns with this change, this change was first introduced at https://github.com/python/cpython/pull/126414, do we need to make it as dummy function for the default build?

Reply:
I don't think the performance overhead of calling this will matter, but if it does, we could replace it with a dummy function that returns a special sentinel value indicating that stats are disabled.

Reply:
Both clang (18.18) and GCC (11.5.0) eliminate the call to `to_bool_fail_kind` when built with `--enable-optimizations` because the result is unused.